src/main/kotlin/com/terraformation/backend/search/SearchResults.kt

```diff
@@ -16,4 +16,61 @@ data class SearchResults(
      * [results] contains the full set of results, this will be null.
      */
     val cursor: String?
-)
+) {
+  /**
+   * Turns nested field values into CRLF-delimited strings suitable for exporting to a CSV file.
+   *
+   * For example, if the client searched for fields `foo.bar` and `foo.oof` and the search results
+   * are
+   *
+   * ```
+   * [
+   *   {
+   *     "foo": [
+   *       { "bar": "bar 1", "oof": "oof 1" },
+   *       { "bar": "bar 2"  }
+   *     ]
+   *   }
+   * ]
+   * ```
+   *
+   * this function will return
+   *
+   * ```
+   * [
+   *   {
+   *     "foo.bar": "bar 1\r\nbar 2",
+   *     "foo.oof": "oof 1"
+   *   }
+   * ]
+   * ```
+   */
+  fun flattenForCsv(): SearchResults {
+    val flattenedResults = results.map { flatten(it) }
+    return SearchResults(flattenedResults, cursor)
+  }
+
+  /** Flattens the values of a single search result. */
+  private fun flatten(result: Map<*, *>): Map<String, String> {
+    return expandNestedFields(result)
+        .groupBy { it.first }
+        .mapValues { (_, pairs) -> pairs.joinToString("\r\n") { it.second } }
+  }
+
+  /** Converts nested search results into a flat list of pairs of field names and values. */
+  private fun expandNestedFields(
+      result: Map<*, *>,
+      prefix: String = ""
+  ): List<Pair<String, String>> {
+    return result.entries.flatMap { (key, value) ->
+      when (value) {
+        is String -> listOf(prefix + key to value)
```

Comment:
just so i remembered correctly.. numbers would be returned as strings in search results.

Reply:
Yes, that's correct, and they'll be formatted according to the locale's number formatting rules (e.g., `"1,234,567"` in English).